### PR TITLE
Feature ETP-272: Boolean Selector without MUI

### PR DESCRIPTION
**Pull request by Sankku**
Removing the use of MUI and improved the styles:

image

### CODE REVIEW

File: packages/MainUI/src/components/FormView/selectors/BooleanSelector.tsx [hunk 1]
```diff
@@ -1,32 +1,64 @@
...
-        />
-      </Box>
-    </FormControl>
-  ),
+  ({ label, readOnly, checked: externalChecked, onChange }) => {
+    const [internalChecked, setInternalChecked] = useState(false);
+    const [isHovered, setIsHovered] = useState(false);
+
+    const isChecked =
+      externalChecked !== undefined ? externalChecked : internalChecked;
```
Comment (LuuchoRocha) [inline packages/MainUI/src/components/FormView/selectors/BooleanSelector.tsx:11]: is this the same as 
```suggestion
    const isChecked = externalChecked || internalChecked?
```

File: packages/MainUI/src/components/FormView/selectors/BooleanSelector.tsx [hunk 2]
```diff
@@ -1,32 +1,64 @@
...
+
+    const borderStyle = {
+      ...styles.checkboxBorder,
+      ...(isHovered ? styles.checkboxBorderHover : {}),
+    };
+
+    return (
+      <div
+        style={styles.checkboxContainer}
+        onMouseEnter={() => setIsHovered(true)}
```
Comment (LuuchoRocha) [inline packages/MainUI/src/components/FormView/selectors/BooleanSelector.tsx:43]: move this out to callbacks

File: packages/MainUI/src/components/FormView/selectors/BooleanSelector.tsx [hunk 3]
```diff
@@ -1,32 +1,64 @@
...
+        onMouseLeave={() => setIsHovered(false)}>
```
Comment (LuuchoRocha) [inline packages/MainUI/src/components/FormView/selectors/BooleanSelector.tsx:44]: same

**Review by LuuchoRocha — CHANGES_REQUESTED** [review]

File: packages/MainUI/src/components/FormView/selectors/BooleanSelector.tsx [hunk 4]
```diff
@@ -1,32 +1,64 @@
...
+        const newChecked = event.target.checked;
+        if (externalChecked === undefined) {
+          setInternalChecked(newChecked);
+        }
+        onChange?.(newChecked);
+      },
+      [externalChecked, onChange],
+    );
+
+    const checkboxStyle = {
```
Comment (tomirodeghiero) [inline packages/MainUI/src/components/FormView/selectors/BooleanSelector.tsx:24]: Using '&&' is better for readability and simplicity
```
const checkboxStyle = {
  ...styles.styledCheckbox,
  isChecked && styles.styledCheckboxChecked,
  readOnly && styles.disabled,
};

const afterStyle = {
  ...styles.styledCheckboxAfter,
  isChecked && styles.styledCheckboxCheckedAfter,
};

const borderStyle = {
  ...styles.checkboxBorder,
  isHovered && styles.checkboxBorderHover,
};
```

File: packages/MainUI/src/components/FormView/selectors/styles.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    backgroundColor: theme.palette.baselineColor.neutral[90],
+    transition: 'height 0.2s ease-in-out',
+  },
+  checkboxBorderHover: {
+    height: '2px',
+  },
+  checkboxLabel: {
+    display: 'inline-block',
+    position: 'relative',
+    paddingLeft: '30px',
```
Comment (tomirodeghiero) [inline packages/MainUI/src/components/FormView/selectors/styles.ts:26]: Move 'px' to 'rem'

File: packages/MainUI/src/components/FormView/selectors/styles.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    opacity: 0,
+    cursor: 'pointer',
+    height: 0,
+    width: 0,
+  },
+  styledCheckbox: {
+    position: 'absolute',
+    top: '2px',
+    left: 0,
+    height: '20px',
```
Comment (tomirodeghiero) [inline packages/MainUI/src/components/FormView/selectors/styles.ts:41]: Move 'px' to 'rem'

File: packages/MainUI/src/components/FormView/selectors/styles.ts [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    width: '20px',
```
Comment (tomirodeghiero) [inline packages/MainUI/src/components/FormView/selectors/styles.ts:42]: Same here :)

File: packages/MainUI/src/components/FormView/selectors/styles.ts [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  },
+  styledCheckboxChecked: {
+    backgroundColor: theme.palette.dynamicColor.main,
+  },
+  styledCheckboxAfter: {
+    content: '""',
+    position: 'absolute',
+    left: '0.5rem',
+    top: '0.25rem',
+    width: '6px',
```
Comment (tomirodeghiero) [inline packages/MainUI/src/components/FormView/selectors/styles.ts:55]: Idem

**Review by tomirodeghiero — CHANGES_REQUESTED** [review]

**Review by LuuchoRocha — APPROVED** [review]

**Pull request merged**